app/src/main/java/com/example/tawazz/icon/exceptions/FailedGettingAllExistingIcons.java

```diff
@@ -0,0 +1,12 @@
+package com.example.tawazz.icon.exceptions;
+
+public class FailedGettingAllExistingIcons extends Exception {
```

Comment:
doesnt end with  `Exception` and also a bit too specific to my taste. maybe just IconFetchException?